captum/optim/_utils/models.py

```diff
@@ -70,22 +71,76 @@ class ReluLayer(nn.Module):
+        super(ReluLayer, self).__init__()
+        self.inplace = inplace
+
     def forward(self, input: torch.Tensor) -> torch.Tensor:
-        return F.relu(input, inplace=True)
+        return F.relu(input, inplace=self.inplace)
 
 
-def replace_layers(model, old_layer=ReluLayer, new_layer=RedirectedReluLayer) -> None:
+def replace_layers(
+    model, layer1, layer2, transfer_vars: bool = False, **kwargs
+) -> None:
     """
-    Replace all target layers with new layers.
-    The most common use case is replacing activation layers with activation layers
-    that can handle gradient flow issues.
+    Replace all target layers with new layers inside the specified model,
+    possibly with the same initialization variables.
+
+    Args:
+        model: (nn.Module): A PyTorch model instance.
+        layer1: (nn.Module): A layer instance that you want to transfer
+            initialization variables from.
+        layer2: (nn.Module): The layer class to create with the variables
+            from of layer1.
+        transfer_vars (bool, optional): Wether or not to try and copy
+            initialization variables from layer1 instances to the replacement
+            layer2 instances.
+        kwargs: (Any, optional): Any additional variables to use when creating
+            the new layer.
     """
 
     for name, child in model._modules.items():
-        if isinstance(child, old_layer):
-            setattr(model, name, new_layer())
+        if isinstance(child, layer1):
+            if transfer_vars:
+                new_layer = _transfer_layer_vars(child, layer2, **kwargs)
+            else:
+                new_layer = layer2(**kwargs)
+            setattr(model, name, new_layer)
         elif child is not None:
-            replace_layers(child, old_layer, new_layer)
+            replace_layers(child, layer1, layer2, transfer_vars, **kwargs)
+
+
+def _transfer_layer_vars(layer1, layer2, **kwargs):
+    """
+    Given a layer instance, create a new layer instance of another class
+    with the same initialization variables as the original layer.
+    Args:
+        layer1: (nn.Module): A layer instance that you want to transfer
+            initialization variables from.
+        layer2: (nn.Module): The layer class to create with the variables
+            from of layer1.
+        kwargs: (Any, optional): Any additional variables to use when creating
+            the new layer.
+    Returns:
+        layer2 instance (nn.Module): An instance of layer2 with the initialization
+            variables that it shares with layer1, and any specified additional
+            initialization variables.
+    """
+
+    l2_vars = list(signature(layer2.__init__).parameters.values())
+    l2_vars = [
+        str(l2_vars[i]).split()[0]
+        for i in range(len(l2_vars))
+        if str(l2_vars[i]) != "self"
+    ]
+    l2_vars = [p.split(":")[0] if ":" in p and "=" not in p else p for p in l2_vars]

```

Comment:
I was just trying this simple case:
```
class C:
    def myfunc(self, a, b:int=10, c=20, d=30):
        pass
l2_vars = list(signature(C().myfunc.__init__).parameters.values())
.....
>> [<Parameter "*args">, <Parameter "**kwargs">]
```
When do you see the output with ':' and '=' symbols ?

Reply:
@NarineK I saw the symbols when replacing MaxPool2d with AvgPool2d layers.

Reply:
Ah, I think in my case it should be:
```
l2_vars = list(signature(C().myfunc).parameters.values())
```
I was calling  __init__.

Reply:
Ah, okay!

Reply:
For this example it looks like we are losing value: 10?
```
class C:
    def myfunc(self, a, b:int=10, c=20, d=30):
        pass

l2_vars = list(signature(C().myfunc).parameters.values())
l2_vars = [
        str(l2_vars[i]).split()[0]
        for i in range(len(l2_vars))
        if str(l2_vars[i]) != "self"
    ]
l2_vars = [p.split(":")[0] if ":" in p and "=" not in p else p for p in l2_vars]

```
.....
['a', 'b', 'c=20', 'd=30']

Reply:
@NarineK Default values that don't overlap between layers can be set via kwargs, or they remain what they are set as when using the `_transfer_layer_vars` function. If the layer we are transfering values from has a `b`, then the value is overridden. If `b` doesn't exist, it will set to 10 in the layer instance.

Reply:
Ah I see and we assume that `shared_vars` and `kwargs` will have matching sizes